torchmultimodal/models/video_gpt.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Tuple
+
+from torch import nn
+from torchmultimodal.models.gpt import (
+    MultimodalGPT,
+    MultimodalTransformerDecoder,
+    RightShift,
+    TransformerDecoder,
+    TransformerDecoderLayer,
+)
+from torchmultimodal.models.video_vqvae import VideoDecoder, VideoEncoder
+from torchmultimodal.models.vqvae import VQVAE
+
+from torchmultimodal.modules.layers.attention import SelfAttention
+from torchmultimodal.modules.layers.position_embedding import (
+    BroadcastedPositionEmbedding,
+)
+
+
+def video_gpt(
+    input_shape: Tuple[int, int, int] = (16, 64, 64),
+    latent_shape: Tuple[int, int, int] = (8, 32, 32),
+    d_model: int = 576,
+    n_head: int = 4,
+    dropout: float = 0.2,
+    attn_dropout: float = 0.3,
+    num_decoder_layers: int = 16,
+    use_gpt_init: bool = True,
+) -> MultimodalGPT:
+    """VideoGPT model
+
+    For the momemnt this is a prototype implementation which may not fully commit to the specifications
```

Comment:
```suggestion
    For the moment this is a prototype implementation which may not fully commit to the specifications
```

Also can you clarify what this comment means? Is it just that we have not run an end-to-end loop with it yet, or is there some refactoring you are anticipating?

Reply:
I will be specific in saying that:
- model arch follows the paper
- parameters are for a particular dataset "BAIR/RobotNet" (the paper has a few other sets of parameters for different datasets: [paper](https://arxiv.org/pdf/2104.10157.pdf) page 13 Table A.1, Table A.2)

I don't think we have to say "we have not run an end-to-end loop" as we haven't claimed it as one of our examples (in `examples/` folder). In other words, we could be implementing components/following parameter specs from a SOTA model without onboarding it as an end-to-end example.

What do you think?